flytekit/core/node_creation.py

```diff
@@ -90,11 +90,17 @@ def sub_wf():
 
     # When compiling, calling the entity will create a node.
     ctx = FlyteContext.current_context()
-    if ctx.compilation_state is not None and ctx.compilation_state.mode == 1:
+    if (ctx.compilation_state is not None and ctx.compilation_state.mode == 1) or (
+        ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION
```

Comment:
@pingsutw , can you leave a note explaining why we need this check?

Reply:
Updated it